bigquery/spec/quickstart_spec.rb

```diff
@@ -0,0 +1,50 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "rspec"
+require "google/cloud"
+
+describe "BigQuery Quickstart" do
+
+  it "creates a new dataset" do
+    # Initialize test objects
+    gcloud_test_client = Google::Cloud.new ENV["GOOGLE_CLOUD_PROJECT"]
+    bigquery_test_client = gcloud_test_client.bigquery
+
+    # Prime BigQuery for test
+    if bigquery_test_client.dataset "my_new_dataset"
+      bigquery_test_client.dataset("my_new_dataset").delete
+    end
+
+    expect(bigquery_test_client.dataset "my_new_dataset").to be nil
+    expect(Google::Cloud).to receive(:new).
+                             with("YOUR_PROJECT_ID").
+                             and_return(gcloud_test_client)
+
+    # Run quickstart
+    expect {
+      load File::expand_path("quickstart.rb")
```

Comment:
Should also specify the full path to the quickstart relative to the spec directory (otherwise this uses the current working directory).  So:

```ruby
load File.expand_path("../quickstart.rb", __dir__)
```